client/editorQueryEvaluator.ts

```diff
@@ -218,4 +281,78 @@ export class EditorQueryEvaluator {
                 return editor;
             });
     }
+
+    private static requestMlxprsWebviewUpdateWithErrorResultsObject(response: ErrorResultsObject): void {
+        if (this.mlxprsWebViewProvider) {
+            const stringResults = this.convertTextResponseToHtml(JSON.stringify(response, null, 2));
+            this.mlxprsWebViewProvider.updateViewContent(stringResults);
+        }
+    }
+
+    private static requestMlxprsWebviewUpdateWithRowsResponse(response: RowsResponse, resultFormat?: ml.RowsResponseFormat): void {
+        if (this.mlxprsWebViewProvider) {
+            let stringResults = '';
+            if (resultFormat === 'json') {
+                stringResults = this.convertTextResponseToHtml(JSON.stringify(response, null, 2));
+            } else if (resultFormat === 'xml') {
+                stringResults = this.convertXmlResponseToHtml(response.toString());
+            } else {
+                stringResults = this.convertTextResponseToHtml(response.toString());
+            }
+            this.mlxprsWebViewProvider.updateViewContent(stringResults);
+        }
+    }
+
+    private static requestMlxprsWebviewUpdateWithRecordArray(response: Record<string, unknown>[], contentType?: string): void {
+        if (this.mlxprsWebViewProvider) {
+            const stringResults =
+                (response as Record<string, unknown>[]).map(record => this.convertRecordToHtml(record, contentType)).join('\n');
+            this.mlxprsWebViewProvider.updateViewContent(stringResults);
+        }
+    }
+
+    private static convertRecordToHtml(record: Record<string, any>, contentType?: string): string {
+        if (record['format'] === 'xml') {
```

Comment:
This all makes sense, but I'm wondering - what is a Record, and what are all the possible keys in it? Like why would "head" be populated but not "value"? Is this all VSCode stuff, or is it stuff we control?

Reply:
I don't have a complete understanding of Record. It is defined in a VSCode library, but it is used to define many of the responses from MarkLogic's Node client. 
These type definitions is an area that could probably benefit from more refactoring and documentation.